### PR TITLE
Additional patterns for ltac2: "as", records, ints, strings

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: doc/changelog/05-tactic-language/16179-tac2pat.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Added:**
+  patterns for Ltac2 matches: ``as``, records and literal ints and strings
```
Comment (jfehrle) [inline doc/changelog/05-tactic-language/16179-tac2pat.rst:2]:
```suggestion
  patterns for Ltac2 matches: ``as``, records and literal integers and strings
```

We haven't been using the abbreviation in the doc.

File: doc/tools/docgram/orderedGrammar [hunk 2]
```diff
@@ -2339,6 +2339,16 @@ tac2rec_fieldexpr: [
...
+tac2rec_fieldpats: [
+| tac2rec_fieldpat ";" OPT tac2rec_fieldpats      (* ltac2 plugin *)
+| tac2rec_fieldpat ";"      (* ltac2 plugin *)
+| tac2rec_fieldpat      (* ltac2 plugin *)
+]
+
+tac2rec_fieldpat: [
+| qualid ":=" tac2pat1      (* ltac2 plugin *)
+]
+
```
Comment (jfehrle) [inline doc/tools/docgram/orderedGrammar:2351]: For readability, please move these lines to just before line 2361 ("ltac2_tactic_atom: [")

**Review by jfehrle — COMMENTED** [review]
Suggestions

**Review by ppedrot — CHANGES_REQUESTED** [review]
This is missing a more comprehensive test suite.

**Comment by SkySkimmer** [comment]
Added some tests

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**